### PR TITLE
Add contextual product more menu

### LINKED ISSUE #35177: [Header] Contextual action buttons – More menu
**User story**

- As a merchant,
- I want to share feedback on the new product editing experience,
- So the WooCommerce development team can continue improving it.

**Design**

5sAIeTRd9Yp7nSCT33BAWz-fi-8921%3A376741

**Acceptance criteria**

- [ ] When the user clicks the ellipsis icon in the top right corner, we show a dropdown menu with several options.
- [ ] The menu looks and behaves exactly as in the Gutenberg editor/FSE.
- [ ] `Share feedback` opens the new CES feedback modal (https://github.com/woocommerce/woocommerce/issues/35122).
- [ ] `Use the classic editor` will let the user switch to the classic editing experience.

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds the product form's more menu and several links.

Note that the CES modal still needs to be completed before it can be hooked up to this component.

Screen Shot 2022-11-01 at 10 42 09 AM

Closes #35177  .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product management experience ( Add new (MVP) )
2. Click the "more" (vertical ellipsis) button in the top right of the header
3. Note that each of the links closes the menu
4. Note that clicking "Use the classic editor" takes you to the classic editing experience
5. Navigate back to the new experience and publish a product
6. Click again on "Use the classic editor"
7. Note that you are taken to the classic editor for the respective published product

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/product-more-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+							{ __( 'Use the classic editor', 'woocommerce' ) }
+						</MenuItem>
+					</>
+				) }
+			</DropdownMenu>
+		</WooHeaderItem>
+	);
+};
+
+registerPlugin( 'woocommerce-product-more-menu', {
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/products/product-more-menu.tsx]: Similar to the this comment https://github.com/woocommerce/woocommerce/pull/35431#discussion_r1013332260 in the product title PR, I don't think this is necessary.

**Review by louwie17 — CHANGES_REQUESTED** [review]
Tests well, and code looks good, just left a similar comment as #35431 in relation to the `registerPlugin` call.

**Comment by joshuatf** [comment]
Thanks for the review @louwie17!  I've rebased this after the product title PR merge and removed the unnecessary plugin registration call. 😄

**Review by louwie17 — APPROVED** [review]
LGTM 🚀 thanks for the rebase and `registerPlugin` update.

**Pull request merged**